File: src/main/java/dog/kaylen/octarine/block/OctarineBlock.java
/*
 * (C) Copyright 2022 Kaylen Dart (https://kaylen.dog) and others.
 * This project is licensed under the GNU General Public License v3.0.
 * See the LICENSE file in the project root for more information.
 */
package dog.kaylen.octarine.block;

import dog.kaylen.octarine.util.Identifiable;
import net.minecraft.block.Block;
import net.minecraft.item.ItemGroup;

public abstract class OctarineBlock extends Block implements Identifiable {
    /**
     * Get the item group of this block.
     *
     * @return The item group of this block
     */
    public abstract ItemGroup getItemGroup();

    public OctarineBlock(Settings settings) {
        super(settings);
    }
}

File: src/main/kotlin/dog/kaylen/octarine/block/OctarineBlock.kt
/*
 * (C) Copyright 2022 Kaylen Dart (https://kaylen.dog) and others.
 * This project is licensed under the GNU General Public License v3.0.
 * See the LICENSE file in the project root for more information.
 */
package dog.kaylen.octarine.block

import dog.kaylen.octarine.registry.Identifiable
import dog.kaylen.octarine.util.identifierOf
import net.minecraft.block.Block

/**
 * A wrapper around the Minecraft block type.
 */
abstract class OctarineBlock(private val key: String, settings: Settings) : Block(settings), Identifiable {
    /**
     * This block's identifier.
     */
    override val identifier get() = identifierOf(this.key)
}
